packages/commons-server/src/libs/server/server.ts

```diff
@@ -521,7 +522,8 @@ export class MockoonServer extends (EventEmitter as new () => TypedEmitter<Serve
                   crudId,
                   servedDatabucket,
                   request,
-                  response
+                  response,
+                  route.responses[0].crudKey
```

Comment:
Use `currentRoute` to get a refreshed route and avoid a server restart when the `crudKey` changes.
```suggestion
                  currentRoute.responses[0].crudKey
```